### PR TITLE
Added a python implementation of the pelican TokenContentIterator

### PR DESCRIPTION
**Pull request by turetske**
This is a python version of TokenContentIterator:

https://github.com/PelicanPlatform/pelican/blob/main/client/acquire_token.go

This is the first step in getting more robust token creation in pelicanfs. This looks for and loads tokens from various environment variables and locations.

### CODE REVIEW

File: src/pelicanfs/token_content_iterator.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+                return token_str
+        except json.JSONDecodeError as err:
+            log.debug(f"Unable to unmarshal file {token_location} as JSON (assuming it is a token instead): {err}")
+            return token_str
+    else:
+        return token_str
+
+
+@dataclass
+class TokenContentIterator:
```
Comment (patrickbrophy) [inline src/pelicanfs/token_content_iterator.py:59]: My first concern is that this class doesn't implement the `__iter__` or `__next__` methods. These methods are part of the Iterator API specified by [Python](https://peps.python.org/pep-0234/).

File: src/pelicanfs/token_content_iterator.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    def next(self) -> Tuple[str, bool]:
+        """
+        Return the next valid token from the configured discovery sources.
+
+        Returns:
+            Tuple[str, bool]: A tuple where the first element is the token string (empty if not found),
+                              and the second element is a boolean indicating success.
+        """
+        while True:
+            match self.method:
```
Comment (patrickbrophy) [inline src/pelicanfs/token_content_iterator.py:143]: This match block is growing quite large. Could you refactor each case into a dedicated private method? Also could you consider using an enum instead of just integers? You could then iterate of over the list of enums. You would have to rework the catch all but going this route may be worth investigating.

File: test/test_open_json.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from unittest.mock import mock_open
+
+import pytest
+
+from pelicanfs.token_content_iterator import get_token_from_file
+
+
```
Comment (patrickbrophy) [inline test/test_open_json.py:22]: Can you add a test case of an empty file?

File: test/test_tci.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    monkeypatch.setattr("os.environ", {}, raising=False)
+    yield
+    # monkeypatch automatically restores os.environ after the test
+
+
+@patch("os.path.exists", return_value=False)
+@patch("os.access", return_value=False)
+def test_explicit_location_unreadable_fallback(mock_access, mock_exists, monkeypatch):
+    iterator = TokenContentIterator(location="/nonexistent/token", name="token_name")
+    iterator.method = 0
```
Comment (patrickbrophy) [inline test/test_tci.py:66]: From a first glance I have no idea what `iterator.method = 0` means, it might be helpful to briefly comment the intent of each test.

File: test/test_tci.py [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    assert success is True
+    assert token == "fallback-token"
+
+
+def test_bearer_token_env_missing_fallback(monkeypatch):
+    iterator = TokenContentIterator(location=None, name="token_name")
+    iterator.method = 1
+    # env is empty because of isolated_env fixture
+    token, success = iterator.next()
+    assert success is False or isinstance(token, str)
```
Comment (patrickbrophy) [inline test/test_tci.py:78]: Consider adding assertion messages to clarify the intent on failure.

File: src/pelicanfs/token_content_iterator.py (hunk 2 above)
Comment (turetske) [reply src/pelicanfs/token_content_iterator.py:143]: Yeah, this was a case of emulating the original switch method.

Comment (turetske) [reply src/pelicanfs/token_content_iterator.py:143]: While the match block is large, I think there really isn't enough meat in there for individual functions. It's a mixture of a repeated block with different error/logging messages.

The Enum part I did implement, however.

**Review by patrickbrophy — APPROVED** [review]
This looks good! LGTM
